Neos.ContentRepository/Classes/Domain/Model/Node.php

```diff
@@ -618,8 +618,8 @@ public function moveBefore(NodeInterface $referenceNode, string $newName = null)
         $name = $newName !== null ? $newName : $this->getName();
         $referenceParentNode = $referenceNode->getParent();
 
-        if ($referenceParentNode !== $this->getParent() && $referenceParentNode->getNode($name) !== null) {
-            throw new NodeExistsException(sprintf('Node with path "%s" already exists.', $name), 1292503468);
+        if ($referenceParentNode !== $this->getParent() && $referenceParentNode->getNode($name) === $this) {
+            throw new NodeExistsException(sprintf('Node with path "%s" already exists.', $this->getPath()), 1292503468);
```

Comment:
I don't understand this. A different node with this name may indeed exist here and would block the path as well, so just checking for `$this` seems insufficient.

Reply:
This is a WIP, sorry.

I'm confused with the unique identity of Nodes in Neos when having multiple dimensions and fallback dimensions.

The `contextPath` is the first unique identity that I found and my assumption is the `node1 === node2` -> `node1->__toString() === node2->__toString()` wich basically compares the `contectPath` of both.

I also looked into what the db has to say about unique keys
<img width="722" alt="Screenshot 2023-12-07 at 9 48 31 AM" src="https://github.com/neos/neos-development-collection/assets/1615332/abc5e050-277e-45fc-bc93-86ae2e89f87f">


I will look into it with @skurfuerst today and hopefully get more insights. Maybe this is an issue with specific configurations and modifications of a customer project.